infrastructure/state.tf

```diff
@@ -15,10 +15,6 @@ terraform {
   backend "azurerm" {}
 
   required_providers {
-    azuread = {
-      source  = "hashicorp/azuread"
```

Comment:
why we are removing this provider, is it defined somewhere else?

Reply:
As far as I can tell, no azuread resources are used in the infrastructure code in this repository - the only reference to azuread is in this file.

The [Terraform provider documentation](https://developer.hashicorp.com/terraform/language/providers#how-to-use-providers) indicates that to be able to use resources from a provider it needs to be declared in required_providers, but as no azuread resources are being used there doesn't seem to be a need to include it.